### PR TITLE
Fix StringSection Assignment

### PR DESCRIPTION
**Pull request by Smit-create**
https://github.com/lcompilers/libasr/issues/11
#1381

### CODE REVIEW

**Comment by certik** [comment]
I think this is fine.

File: tests/reference/llvm-class_02-82c2f9c.stdout [hunk 1]
```diff
@@ -75,7 +75,7 @@ return:                                           ; preds = %.entry
   ret void
 }
 
-; Function Attrs: nounwind readnone speculatable willreturn
+; Function Attrs: nofree nosync nounwind readnone speculatable willreturn
```
Comment (Smit-create) [inline tests/reference/llvm-class_02-82c2f9c.stdout:78]: I have upgraded my LLVM version to 14 and this gets updated automatically.

Comment (Smit-create) [reply tests/reference/llvm-class_02-82c2f9c.stdout]: Reverted this change by updating tests with LLVM 11.

Comment (certik) [reply tests/reference/llvm-class_02-82c2f9c.stdout]: Yes, you have to use exactly LLVM 11 to update LLVM tests.

**Review by certik — APPROVED** [review]
I think this looks good.

**Pull request merged**